sycl/test/esimd/gather_scatter.cpp

```diff
@@ -51,3 +54,12 @@ void kernel(accessor<uint16_t, 1, access::mode::read_write,
 
   scatter<uint16_t, 32>(buf, offsets, v0);
 }
+
+void conv_kernel(accessor<uint16_t, 1, access::mode::read_write,
+                          access::target::device> &buf) SYCL_ESIMD_FUNCTION {
+
+  // Make sure we can pass offsets as a scalar.
+  simd<uint16_t, 32> v0 = gather<uint16_t, 32>(buf, 0);
+
+  scatter<uint16_t, 32>(buf, 0, v0);
```

Comment:
Late comment: having this for compile-only test is ok. For runtime test the scatter of N elements to same memory should be Ok only for N=1, otherwise it is UB.